core/src/main/scala/kafka/log/LocalLog.scala

```diff
@@ -517,14 +558,16 @@ class LocalLog(@volatile private var _dir: File,
       debug(s"Truncate and start at offset $newOffset")
       checkIfMemoryMappedBufferClosed()
       val segmentsToDelete = List[LogSegment]() ++ segments.values
-      removeAndDeleteSegments(segmentsToDelete, asyncDelete = true, LogTruncation(this))
-      segments.add(LogSegment.open(dir,
-        baseOffset = newOffset,
-        config = config,
-        time = time,
-        initFileSize = config.initFileSize,
-        preallocate = config.preallocate))
+      if (segmentsToDelete.size > 1)
```

Comment:
Would it be better to combine the two ifs together to sth like the following? It would also be useful to add a comment why we want do special treatment for the last segment.

```
      if (segmentsToDelete.nonEmpty) {
        removeAndDeleteSegments(segmentsToDelete.dropRight(1), asyncDelete = true, LogTruncation(this))
        createAndDeleteSegment(newOffset, segmentsToDelete.last, asyncDelete = true, LogTruncation(this))
     }
```

Reply:
Makes sense. removeAndDeleteSegments has a check for empty lists, so it makes sense to combine the if statements.